x-pack/legacy/plugins/ml/public/plugin.ts

```diff
@@ -19,6 +19,7 @@ export class MlPlugin implements Plugin<Setup, Start> {
           element: params.element,
           appBasePath: params.appBasePath,
           onAppLeave: params.onAppLeave,
+          history: params.history,
```

Comment:
I see you added `history` to the ML plugin here. We're using `HashRouter` (see [here](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/ml/public/application/routing/router.tsx#L50)) and the `useHistory()` hook of `react-router-dom`. Do we need to add `history` to `HashRouter`?

Reply:
The `transform` plugin is also making use of `useHistory()`. Is this something you want to look into as part of this PR or should we pick it up as a follow up?

Reply:
I believe the `useHistory()` hook should could continue to work, you'll just want to wire up this history instance to your HashRouter by passing this object into the `history` prop once this PR is merged.

That said, I would recommend switching to the regular BrowserRouter once on the New Platform. You will need to setup redirects to make sure any bookmarks using the hash URLs don't break.